pkg/planner/core/casetest/vectorsearch/testdata/ann_index_suite_in.json

```diff
@@ -44,5 +44,41 @@
       "explain select * from t1 where store_id between 80 and 120 order by vec_cosine_distance(vec, '[1,1,1]') limit 1",
       "explain select * from t1 partition (p0) order by vec_cosine_distance(vec, '[1,1,1]') limit 1"
     ]
+  },
+  {
+    "name": "TestVectorSearchWithPKAuto",
+    "cases": [
+      "explain select id from t1",
+
+      "explain select id from t1 order by vec_cosine_distance(vec, '[1,1,1]') limit 10",
```

Comment:
i notice that currently we only deal with the case of vec_func in the order by clause, is it possible when user encounter sql like below:
```
explain select id from t1 having vec_cosine_distance(vec, '[1,1,1]') > 1 order by vec_cosine_distance(vec, '[1,1,1]') limit 10
```

Reply:
I test this sql and get `[planner:1054]Unknown column 'vec' in 'having clause'`.
Maybe using `select id,vec` is the right approach? because `having` can only get the result of `select`.
and the `select id,vec` case is:
```sql
[Projection_7 10.00 root  test.t1.id, test.t1.vec]
[└─TopN_10 10.00 root  Column#8, offset:0, count:10]
[  └─TableReader_24 10.00 root  MppVersion: 2, data:ExchangeSender_23]
[    └─ExchangeSender_23 10.00 mpp[tiflash]  ExchangeType: PassThrough]
[      └─TopN_22 10.00 mpp[tiflash]  Column#8, offset:0, count:10]
[        └─Projection_21 10.00 mpp[tiflash]  test.t1.id, test.t1.vec, vec_cosine_distance(test.t1.vec, [1,1,1])->Column#8]
[          └─Selection_19 4800.00 mpp[tiflash]  gt(vec_cosine_distance(test.t1.vec, [1,1,1]), 1)]
[            └─TableFullScan_18 6000.00 mpp[tiflash] table:t1 pushed down filter:empty, keep order:false, stats:partial[vec:unInitialized]]
```

Reply:
how about 
```
explain select id from t1 group by id, vec_cosine_distance(vec, '[1,1,1]') having vec_cosine_distance(vec, '[1,1,1]') > 1 order by vec_cosine_distance(vec, '[1,1,1]') limit 10
```

Reply:
That will get same result. I think it isn't the impact of this PR